front/app/modules/commercial/insights/services/insightsViews.ts

```diff
@@ -0,0 +1,76 @@
+import { API_PATH } from 'containers/App/constants';
+import streams, { IStreamParams } from 'utils/streams';
+import { IRelationship } from 'typings';
+
+export interface IInsightsViewData {
+  id: string;
+  type: string;
+  attributes: {
+    name: string;
+  };
+  relationships?: {
+    scope: {
+      data: IRelationship;
+    };
+  };
+}
+
+export interface IInsightsView {
+  data: IInsightsViewData;
+}
+
+export interface IInsightsViews {
+  data: IInsightsViewData[];
+}
+
+export interface IInsightsView {
+  data: IInsightsViewData;
+}
+
+interface IInsightsViewObject {
+  scope_id: string;
+  name: string;
+}
+
+const insightsViewsEndpoint = 'insights/views';
+
+export function insightsViewsStream(streamParams: IStreamParams | null = null) {
+  return streams.get<IInsightsViews>({
+    apiEndpoint: `${API_PATH}/${insightsViewsEndpoint}`,
+    ...streamParams,
+  });
+}
+
+export function insightsViewStream(
+  insightsViewId: string,
+  streamParams: IStreamParams | null = null
+) {
+  return streams.get<IInsightsView>({
+    apiEndpoint: `${API_PATH}/${insightsViewsEndpoint}/${insightsViewId}`,
+    ...streamParams,
+  });
+}
+
+export function addInsightsView(object: IInsightsViewObject) {
+  return streams.add<IInsightsView>(`${API_PATH}/${insightsViewsEndpoint}`, {
+    view: object,
+  });
+}
+
+export function updateInsightsView(
```

Comment:
To be verified, but I this the scope cannot be changed, if so the typings should prevent from trying to update it